docs/en_US/NAS/retiarii/WriteTrainer.rst

```diff
@@ -9,53 +9,87 @@ Trainers are necessary to evaluate the performance of new explored models. In NA
 Classic trainers
 ----------------
 
-All classic trainers need to inherit ``nni.retiarii.trainer.BaseTrainer``, implement the ``fit`` method and decorated with ``@register_trainer`` if it is intended to be used together with Retiarii. The decorator serialize the trainer that is used and its argument to fit for the requirements of NNI.
+It's recommended to use write training code in PyTorch-Lightning style, that is, to write a LightningModule that defines all elements needed for training (e.g., loss function, optimizer) and to define a trainer that takes (optional) dataloaders to execute the training. You can refer to `document of PyTorch-lightning <https://pytorch-lightning.readthedocs.io/>` for details.
 
-The init function of trainer should take model as its first argument, and the rest of the arguments should be named (``*args`` and ``**kwargs`` may not work as expected) and JSON serializable. This means, currently, passing a complex object like ``torchvision.datasets.ImageNet()`` is not supported. Trainer should use NNI standard API to communicate with tuning algorithms. This includes ``nni.report_intermediate_result`` for periodical metrics and ``nni.report_final_result`` for final metrics.
+In pratice, writing a new training module in NNI should inherit ``nni.retiarii.trainer.pytorch.lightning.LightningModule``, which has a ``set_model`` that will be called after ``__init__`` to save the candidate model (generated by strategy) as ``self.model``. The rest of the process (like ``training_step``) should be the same as writing any other lightning module. Trainers should also communicate with strategies via two API calls (``nni.report_intermediate_result`` for periodical metrics and ``nni.report_final_result`` for final metrics), added in ``on_validation_epoch_end`` and ``teardown`` respectively. 
 
 An example is as follows:
 
 .. code-block::python
 
-    from nni.retiarii import register_trainer
-    from nni.retiarii.trainer import BaseTrainer
+    from nni.retiarii.trainer.pytorch.lightning import LightningModule  # please import this one
 
-    @register_trainer
-    class MnistTrainer(BaseTrainer):
-        def __init__(self, model, optimizer_class_name='SGD', learning_rate=0.1):
+    @blackbox_module
+    class AutoEncoder(LightningModule):
```

Comment:
since we do not define model in `LightningModule`, if we still use the name `LightningModule`, it may be misleading. we can use more understandable name, for example, `BaseTrainer`, `TrainingModule`, etc.